### ISSUE TITLE
flow service: add support for "real" urls

### ISSUE THREAD

**Issue by alex3683** [LaxarJS/laxar#366]
Up to now LaxarJS only supports routing based on url anchors. The new routing library `page.js` already supports real url based routing and the methods of the flow service already try to abstract from the underyling method of routing. Nonetheless in reality hash-based routing is still expected and a `constructAnchor()` method is explicitly provided for this. The `constructAbsoluteUrl()` also uses this method to create absolute urls for links.

Long story short: `constructAnchor()` should be removed and `constructAbsoluteUrl()` should create a url that fits the configured type of routing. `constructPath()` should be fine and create urls that fit either type of routing. Nevertheless there is no need that this method is public, since it only makes any sense if the underlying routing library is known and publicly accessible.

**Comment by x1B** [comment]
Implemented on `master` (`v2.0`).

### BREAKING CHANGE

The API of the widget injection `axFlowService` has changed. *Previously* there were three methods:

 - `constructPath( placeOrTarget, params )` to obtain the part of the a place's URL that is local to the configured routing setup (e.g. the stuff after `#!` if using hashbang URLs).
 - `constructAbsoluteUrl( placeOrTarget, params )` to calculate an absolute URI including path and hashbang-suffix, for use in `a` tags. Only worked correctly if hashbang URLs were used.
 - `constructAnchor( placeOrTarget, params )` to calculate the hash-based link for use in `a` tags, without path prefix. Only useful if hashbang URLs were used.

The whole point of the declarative flow is that widgets should be portable across different routing scenarios. So, the same widgets should work equally well independent of wether HTML5 URLs or hash fragments are used for routing. Also, in REST-philosophy, URLs should be opaque: Widgets should not try to build URLs from a path suffix. 

For this reason, *now only one* method remains:

 - `absoluteUrl( placeOrTarget, params )` calculates an absolute URL for the given place that should work independent of the current router setup. Widgets should use this method instead of any of the above methods. The method `constructAbsoluteUrl` is kept as an alias since it already had the same purpose (but only worked with hash-based routing).

**Comment by x1B** [comment]
_Updated upgrade information_ after #387
